src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1814,6 +1817,27 @@ void Partitioner::optimize(const std::string& func_name) {
         std::vector<ov::npuw::patterns::opt::Context::PPtr> to_remove;
         std::set<std::size_t> to_remove_idx;
 
+        // Update params and closures after convolution to matmul passes
+        for (auto&& p : ctx.closures_to_reshape_add) {
+            new_params.push_back(p);
+        }
+        for (auto&& p : ctx.closures_to_reshape_remove) {
+            auto param_idx = f._model->get_parameter_index(p);
+            to_remove.push_back(p);
+            to_remove_idx.insert(param_idx);
+            auto closure_idx = param_idx - f._param_offset;
+            for (std::size_t f_idx = 0; f_idx < func_group.refs.size(); ++f_idx) {
+                auto& funcall = func_group.refs[f_idx].get();
+                // Update and move closure according to the newly added parameter
+                // FIXME: rewrite with functional LazyTensors
+                // !!! FIXME !!! not tested yet. Won't work with original params and not consts!
+                funcall._lazy_closure[closure_idx].update(TransformType::RESHAPE, std::monostate{});
```

Comment:
that's not really a reshape as you're not reshaping anything. I'd actually make it a reshape (that is, a generic one - accepting a shape tensor to reshape into), because why not

Reply:
Removed